compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -865,7 +865,48 @@ OMR::Power::TreeEvaluator::PassThroughEvaluator(TR::Node *node, TR::CodeGenerato
 TR::Register*
 OMR::Power::TreeEvaluator::mAnyTrueEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return vloadEvaluator(node, cg);
+   TR::Node *inputNode = node->getFirstChild();
+   TR::Node *maskNode = node->getOpCode().isVectorMasked() ? node->getSecondChild() : NULL;
+
+   TR::Register *inputReg = cg->evaluate(inputNode);
+   TR::Register *maskReg = maskNode ? cg->evaluate(maskNode) : NULL;
+
+   TR::Register *resultReg = cg->allocateRegister(TR_GPR);
+   TR::Register *temp = cg->allocateRegister(TR_VRF);
+   TR::Register *zeroReg = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(resultReg);
+
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::vspltisw, node, zeroReg, 0);
+
+   if (maskReg)
+      generateTrg1Src3Instruction(cg, TR::InstOpCode::xxsel, node, temp, zeroReg, inputReg, maskReg);
+
+   //count leading zeroes for each word element and take sum
+   //if input is all 0's (i.e.: anyTrue is FALSE), each word element will have 32 leading zeroes, and the sum 
+   //will be 32 + 32 + 32 + 32 = 128 = 2^7
+   generateTrg1Src1Instruction(cg, OMR::InstOpCode::vclzw, node, temp, (maskReg ? temp : inputReg));
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::vsumsws, node, temp, temp, zeroReg);
+
+   //move result to GPR
+   if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P9))
+      generateTrg1Src1Instruction(cg, TR::InstOpCode::mfvsrld, node, resultReg, temp);
+   else
+      {
+      generateTrg1Src2ImmInstruction(cg, TR::InstOpCode::xxpermdi, node, temp, temp, zeroReg, 3); //move sum to upper doubleword element of tempRes
+      generateTrg1Src1Instruction(cg, TR::InstOpCode::mfvsrd, node, resultReg, temp);
+      }
+
+   //shift right 7 bits and flip least significant bit to get result (return 1 if true, 0 if false)
+   generateTrg1Src1ImmInstruction(cg, OMR::InstOpCode::sradi, node, resultReg, resultReg, 7);
+   generateTrg1Src2Instruction(cg, OMR::InstOpCode::nor, node, resultReg, resultReg, resultReg);
```

Comment:
`nor` seems not the right instruction.  given the input being 0 or 1, `nor ` result is either -1 or -2 in two's complement representation (0xFFFFFFFFFFFFFFFF or 0xFFFFFFFFFFFFFFFE). you are flipping every bit in register.
you should use `xori resReg, resReg, 1` (only flipping the last bit)